text/0000-handleCut.md

```diff
@@ -0,0 +1,51 @@
+# Summary
+
+Add a handleCut editor property to EditorView to allow for custom cut behavior.
+
+# Motivation
+
+The handleCut property would allow others to apply custom behavior to cut events while leveraging prosemirror's built in cross browser support for the clipboard api. One use case is for not deleting the selected content on cut but instead adding a mark to the content. For example: tracking changes made to a document by striking a line through deleted content instead of removing it.
+
+A handleCut property would allow a custom handler to apply any transactions to the document instead of the default transaction done by prosemirror.
+
+It is currently possible to use a handleDOMEvents.cut handler to override the default prosemirror behavior but it requires the user to write their own cross-browser clipboard support.
+
+# Guide-level explanation
+
+A new property would be added to EditorProps named handleCut. It would behave similarly to other handle properties (handlePaste, handleDrop, etc). If the handleCut function returns true then prosemirror's default behavior is prevented. The handleCut function can carry out its own transaction instead of prosemirror's default behavior of deleting the current selection.
```

Comment:
Does it also call `Event#preventDefault()` (I believe this is the standard behaviour for returning `true` from the other `handleXXX` methods)?